client/src/Tools/_framework/ToolPanels/MoveContentToFolder.tsx

```diff
@@ -0,0 +1,200 @@
+    }
+  }
+
+  const initialRef = React.useRef(null);
+
+  const fetcher = useFetcher();
+
+  return (
+    <>
+      <Modal
+        isOpen={isOpen}
+        onClose={onClose}
+        size="xl"
+        initialFocusRef={initialRef}
+      >
+        <ModalOverlay />
+        <ModalContent>
+          <ModalHeader>
+            <Heading>Move "{contentName}" to folder:</Heading>
+            <HStack>
+              {activeView.folder ? (
+                <>
+                  <IconButton
+                    icon={<ArrowBackIcon />}
+                    aria-label="Back"
+                    onClick={() =>
+                      updateActiveView(activeView.folder!.parentFolderId)
+                    }
+                  />
+                  <Text>{activeView.folder!.name}</Text>
+                </>
+              ) : (
+                <Text>My Activities</Text>
+              )}
+            </HStack>
+          </ModalHeader>
+          <ModalCloseButton />
+          <ModalBody>
+            <SimpleGrid columns={1} spacing={0}>
+              {activeView.contents.length === 0 ? (
+                <Text as="i">This folder is empty.</Text>
+              ) : (
+                activeView.contents.map((content) => (
+                  <Button
+                    key={`moveContentItem${content.id}`}
+                    variant="outline"
+                    width="500px"
+                    height="2em"
+                    onClick={() => {
+                      if (content.isFolder && content.id !== id)
+                        updateActiveView(content.id);
+                    }}
+                    isDisabled={!content.isFolder || content.id === id}
+                  >
+                    <Text width="50%" marginTop="auto" marginBottom="auto">
+                      {content.name}
+                    </Text>
+                  </Button>
+                ))
+              )}
+            </SimpleGrid>
+          </ModalBody>
+
+          <ModalFooter>
+            <Button ref={initialRef} mr={3} onClick={onClose}>
+              Cancel
+            </Button>
+            <Button
+              onClick={() => {
+                fetcher.submit(
+                  {
+                    _action: "Move",
+                    id,
+                    folderId: activeView.folder ? activeView.folder.id : null,
+                    desiredPosition: activeView.contents.length, // place it as the last item
+                  },
+                  { method: "post" },
+                );
+                onClose();
+              }}
+              // Is disabled if the content is already in this folder
+              isDisabled={
+                (activeView.folder === null && parentId === null) ||
+                (activeView.folder !== null &&
+                  parentId !== null &&
+                  activeView.folder.id === parentId)
+              }
+            >
+              Move to {activeView.folder?.name ?? "My Activities"}
```

Comment:
To keep this in check with a long name, you can use the `noOfLines` attribute of `<Text>` (along with a `<ToolTip>` to display the whole name:
```
<Tooltip
  label={`Move to ${activeView.folder?.name ?? "My Activities"}`}
>
  <Text noOfLines={1}>
    Move to {activeView.folder?.name ?? "My Activities"}
  </Text>
</Tooltip>
```